core/src/main/java/org/opensearch/remote/metadata/client/SdkClient.java

```diff
@@ -289,9 +289,20 @@ private void validateTenantId(String tenantId) {
     }
 
     /**
-     * Throw exception if tenantId is null for any bulk request and multitenancy is enabled
+     * Throw exception if tenantId is null and multitenancy is enabled. For specific case, like when {@link SdkClient#isGlobalResource(String, String)}
+     * method been invoked, the tenant id in the request is global tenant id, we shouldn't block these requests.
      * @param tenantId The tenantId from the request
      */
+    private void validateTenantIdNotNull(String tenantId) {
+        if (Boolean.TRUE.equals(isMultiTenancyEnabled) && Strings.isNullOrEmpty(tenantId)) {
+            throw new IllegalArgumentException("A tenant ID is required when multitenancy is enabled.");
+        }
+    }
```

Comment:
We already have (see line 281) a `validateTenantId()` method that checks whether the tenant ID is null (and additionally checks if it matches the global tenant id.)   

This new `validateTenantIdNotNull()` method duplicates most of that logic.

My suggestion:  modify the `validateTenantId()` method on line 281 to take a second `boolean` argument `allowGlobalTenantMatch` (or `forbidGlobalTenantMatch`).  Then only test the `tenantId.equals(globalTenantId)` condition based on that boolean.  (So for the put, update, delete, and search you would not allow tenant ID to match global, but for get you would use the opposite boolean.)